Added [codetally] to shields project.

triggerman722 opened the pull request:
Looking to add codetally to the shields project.

triggerman722 commented:
Prior build failure appeared to be due to call to github(?). Retrying.

triggerman722 commented:
#1054 would be awesome in addition to this merge. Thanks!

paulmelnikow commented on service-tests/codetally.js:
```diff
@@ -0,0 +1,23 @@
+  .get('/triggerman722/colorstrap.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: Joi.equal('codetally'),
+    value: Joi.string().regex(/\b\d+(?:.\d+)?/)
```
Could you add a comment indicating the intent of this regex?

paulmelnikow commented on server.js:
```diff
@@ -1090,6 +1090,36 @@ cache(function(data, match, sendBadge, request) {
+    }
+    try {
+      var data = JSON.parse(buffer);
+      badgeData.text[1] = " " + data.currency_sign + " " + data.amount + " " + data.multiplier;
```
Could you omit the space between the currency sign and the amount? **$2.50** instead of **$ 2.50**.

paulmelnikow commented on logo/codetally-white.svg:
```diff
@@ -0,0 +1,9 @@
+    <rect x="6" y="3" width="1" height="7"/>
+    <line x1="0.25" y1="9.75" x2="6.75" y2="3.25" stroke-width="1" stroke="#fff" />
+  </g>
+</svg>
```
There isn't consensus about how to handle showing vendor logos, particularly by default. The design goal of shields was to display clean, consistent badges. The social badges have logos, though there are only a couple other badges with logos.

Would you mind removing it until a consensus is reached? It would expedite getting this merged.

triggerman722 replied:
Thank you Paul. I will update to remove the logo, space and provide testing comments this and re-submit.

triggerman722 commented:
Hi Paul,

I've made the 3 requested changes:
1. Removed the space between $ and the value
2. Removed the logo
3. Added a comment explaining the regex in the test.

Note: the travis tests fail, however that is due to a 403 from github which occurs periodically due to rate limiting (i.e. not specific to this change).

paulmelnikow commented:
Thanks! I just re-ran the Travis build and one of the new Codetally tests failed. Ran it one more time, and it passed. Hopefully a fluke!

paulmelnikow commented:
Just caught another failure on that test in a different CI build.

I timed the request by running `time curl http://www.codetally.com/formattedshield/triggerman722/colorstrap`. Oddly the first time I ran it, the request took almost two seconds. After that, it took about 100–200 ms.

Could there be some caching on the server that makes it work on subsequent requests, but not the first one?

It seems to me that, to keep this badge, we need their API to respond faster than two seconds…

Could you raise the issue upstream?

triggerman722 commented:
Hi Paul,

Codetally is on heroku and yesterday I upgraded my plan there to avoid this problem. On Heroku's cheap plans (which is what Codetally was on when you saw this problem) sites with low volume get put to sleep, then fire up on first request. That is why you had 2s then 100-200ms. From now on, you should have neither problem (i.e. no more initial lag and individual hits should be faster as well.).

If we see a continued trend of poor performance, please log an issue in the codetally_core project - https://github.com/Codetally/codetally_core/issues/new.

Thanks

paulmelnikow commented:
Aha! It was a Hobby dyno. Thanks for addressing that.
